Source/NSSortDescriptor.m

```diff
@@ -167,11 +186,52 @@ + (id) sortDescriptorWithKey: (NSString *)aKey
                                       selector: aSelector]);
 }
 
++ (id)sortDescriptorWithKey: (NSString *)key 
+                  ascending: (BOOL)ascending 
+                 comparator: (NSComparator)cmptr
+{
+  return AUTORELEASE([[self alloc] initWithKey: key
+                                     ascending: ascending
+                                    comparator: cmptr]);
+  
+}
+
 - (id) initWithKey: (NSString *) key ascending: (BOOL) ascending
 {
   return [self initWithKey: key ascending: ascending selector: NULL];
 }
 
+- (id) initWithKey: (NSString *) key
+         ascending: (BOOL) ascending
+        comparator: (NSComparator) cmptr
+{
+  if ([self init])
+    {
+      if (key == nil)
+        {
+          [NSException raise: NSInvalidArgumentException
+                      format: @"%@", _(@"Passed nil key when initializing "
+                                       @"an NSSortDescriptor.")];
+        }
+      if (cmptr == NULL)
+        {
+          [NSException raise: NSInvalidArgumentException
+                      format: @"%@", _(@"Passed NULL comparator when initializing "
+                                       @"an NSSortDescriptor.")];
+        }
+
+      ASSIGN(_key, key);
+      _ascending = ascending;
+      _comparator = cmptr;
```

Comment:
Shouldn't we somehow retain the NSBlock used for comparison? And if so, we will need a matching release in the dealloc method.

Reply:
Done